homeassistant/components/xiaomi_miio/alarm_control_panel.py

```diff
@@ -0,0 +1,162 @@
+"""Support for Xiomi Gateway alarm control panels."""
+
+from functools import partial
+import logging
+
+from miio import DeviceException
+
+from homeassistant.components.alarm_control_panel import (
+    SUPPORT_ALARM_ARM_AWAY,
+    AlarmControlPanel,
+)
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMING,
+    STATE_ALARM_DISARMED,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ["python-miio==0.5.0.1"]
+
+XIAOMI_STATE_ARMED_VALUE = "on"
+XIAOMI_STATE_DISARMED_VALUE = "off"
+XIAOMI_STATE_ARMING_VALUE = "oning"
+XIAOMI_SUCCESS = ["ok"]
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Hue lights from a config entry."""
+    devices = []
+    gateway = hass.data[DOMAIN][config_entry.entry_id]
+    device = XiaomiGatewayAlarm(
+        gateway,
+        config_entry.title + " Alarm",
+        config_entry.data.get("model"),
+        config_entry.data.get("mac"),
+        config_entry.data.get("gateway_id"),
+    )
+    devices.append(device)
+    async_add_entities(devices)
+
+
+class XiaomiGatewayAlarm(AlarmControlPanel):
+    """Representation of the XiaomiGatewayAlarm."""
+
+    def __init__(self, gateway_device, gateway_name, model, mac_address, device_id):
+        """Initialize the entity."""
+        self._gateway = gateway_device
+        self._name = gateway_name
+        self._skip_update = False
+        self._device_id = device_id
```

Comment:
What kind of id is the device id?

Reply:
that is the unique_id of the gateway device, this is later used in the `device_info` property to define the identifiers to link it to the gateway device.
I will rename to make it clearer